dockets/error_queue.py

```diff
@@ -57,7 +58,11 @@ def requeue_all_errors(self):
             self.requeue_error(error_id)
 
     def delete_error(self, error_id):
-        self.queue.delete(self.error(error_id)['envelope'])
+        error = self.error(error_id)
```

Comment:
It sounds like what you want is to no longer consider not having `error['envelope']` to be exceptional, in which case I'd recommend just doing a regular `in` check.